### DISCUSSION TITLE
Should WebFinger redirect if the user has moved instance?

### DISCUSSION THREAD

**Discussion by voltagex** [mastodon/mastodon#20558] [Q&A]
Edit: digging further - I believe that WebFinger should be redirecting in the first place.

tl;dr: Clicking 'follow' on the authorize_interaction page leads to a 403 forbidden page if the user has moved instance.

Scenario:
User A uses Fedifinder or something similar to find User B via @userb@mastodon.example.org. User A happens to also be on mastodon.example.org

User B has moved to @userb@example.com and redirected their account.

The authorize_interaction URL is generated as follows: https://mastodon.example.org/authorize_interaction?uri=https://mastodon.example.org/@userb

Clicking follow POSTs to  https://mastodon.example.org/authorize_interaction which returns 403. Wouldn't it be better to return 302 and redirect to the new instance in this case?
